src/sc2laddercore/LadderManager.cpp

```diff
@@ -264,30 +264,59 @@ std::string LadderManager::GetBotCommandLine(const BotConfig &AgentConfig, int G
 	{
 	case Python:
 	{
-		OutCmdLine = "python " + AgentConfig.FileName;
+		OutCmdLine = Config->GetValue("PythonBinary") + " " + AgentConfig.FileName;
+		OutCmdLine += " --GamePort " + std::to_string(GamePort) + " --StartPort " + std::to_string(StartPort) + " --LadderServer 127.0.0.1 --OpponentId " + OpponentId;
 		break;
 	}
+	case Wine:
+	{
+		OutCmdLine = "wine " + AgentConfig.FileName;
+		OutCmdLine += " --GamePort " + std::to_string(GamePort) + " --StartPort " + std::to_string(StartPort) + " --LadderServer 127.0.0.1 --OpponentId " + OpponentId;
+		break;
+	}
+    case Mono:
+    {
+        OutCmdLine = "mono " + AgentConfig.FileName;
+        OutCmdLine += " --GamePort " + std::to_string(GamePort) + " --StartPort " + std::to_string(StartPort) + " --LadderServer 127.0.0.1 --OpponentId " + OpponentId;
+        break;
+    }
+    case CommandCenterLinux:
+    {
+        OutCmdLine = "wine " + Config->GetValue("CommandCenterPath") + " --ConfigFile " + AgentConfig.FileName;
+        OutCmdLine += " --GamePort " + std::to_string(GamePort) + " --StartPort " + std::to_string(StartPort) + " --LadderServer 127.0.0.1 --OpponentId " + OpponentId;
```

Comment:
Now it is not enoufg spaces here and in several other parts of the switch in this file.